src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,57 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUser(user);
+        checkLogin(user);
+        checkPassword(user);
+        checkAge(user);
+        return storageDao.add(user);
+    }
+
+    private void checkUser(User user) {
+        if (user == null) {
+            throw new RegistrationException("User can't be null!");
+        }
+    }
+
+    private void checkLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null!");
+        }
+        if (user.getLogin().isEmpty()) {
+            throw new RegistrationException("Login can't be empty!");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("Login " + user.getLogin() + " are exist!");
```

Comment:
```suggestion
            throw new RegistrationException("Login " + user.getLogin() + " is exist!");
```

Reply:
I'll be more careful with grammar. Thank you